Various improvements

p-kozlowski opened the pull request:
- on-demand env cloning
- TraceId
- go fmt
- updated vendored gmock

przemyslaw-dobrowolski-cl committed:
Enable gohan server during bash completion testing of gohan client

p-kozlowski commented:
@przemyslaw-dobrowolski-cl should we merge this PR?

przemyslaw-dobrowolski-cl reviewed (approved):
all commits were cherry-picked. this PR can be closed now.

Commit:
Merge pull request #505 from michalzurawski/golang_extensions

Golang extensions

p-kozlowski committed:
Propagate error code from exceptions thrown by extensions

p-kozlowski committed:
Fail ETCD tests early: do not allow errors to propagate

skizhak committed:
templates to support passing package, language

Commit:
Remove Priority type

Since Priority is just an alias of int and there is no additional
restrictions on top of int, we simply use int here.

nati committed:
Merge pull request #523 from cloudwan/fix_priority_type

Remove Priority type

nati committed:
Merge pull request #524 from skizhak/update-templates

generate templates to support passing in package, language

nati committed:
Merge branch 'master' into fix_error_code

Commit:
Merge pull request #522 from p-kozlowski/fix_error_code

Propagate error code from exceptions thrown by extensions

przemyslaw-dobrowolski-cl committed:
Golang framework improvements:
 - added LockFetch LockList
 - added multi-suite aggregating ginkgo reporter
 - added Http to golang exts
 - added Config to golang tests
 - added load binaries from schemas automatically during go tests
 - added NewError constructor
 - added informational message about missing raw types
 - added db delete and create
 - added test for handlers not invoked
 - added IAuth
 - added automatic transaction retry in goext/Within
 - fixed Within implementation
 - fixed error code handling
 - fixed test isolation in golang tests
 - fixed a bug in environment clone: it should correctly clone bound implementations
 - fixed a bug with missing Start of Environment
 - fixed "resource not found" to be strongly typed
 - fixed invalid clone of runtime types in clone (deepcopy does not work for reflect.Type)
 - remove global variables from golang env
 - remove IEnvironmentRef since is breaks environment mocking

p-kozlowski committed:
Timeouts for golang extensions

- pass cancelable context to golang handlers
- cancel the context on extension execution timeout
- pass context to sync.Watch

p-kozlowski committed:
Add resource to context in GetSingleResource

przemyslaw-dobrowolski-cl committed:
Merge pull request #518 from przemyslaw-dobrowolski-cl/fix_bash_completion_tests

Enable gohan server during bash completion testing of gohan client

jkmar committed:
moved NewUUID and Config to separate files

michalzurawski committed:
Fixed trigger event

p-kozlowski committed:
Type-safe casting of context[transaction]

Commit:
ISchemas expose information about schema relations

Commit:
Merge pull request #527 from przemyslaw-dobrowolski-cl/golang_extensions

Golang framework improvements

p-kozlowski committed:
Merge branch 'resource_in_context' into golang_extensions

p-kozlowski committed:
Merge branch 'goext_timeouts' into golang_extensions

p-kozlowski committed:
Merge branch 'ctx_transaction_fix' into golang_extensions

p-kozlowski committed:
Merge branch 'golang-fixed-handle-event' into golang_extensions

Commit:
Do not return ErrResourceNotFound during delete if requested resource doesn't exist
